src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/Utils.cs

```diff
@@ -396,7 +396,7 @@ internal static long Pump(Stream input, Stream output)
             int numBytes;
             long totalBytes = 0;
 
-            while ((numBytes = input.Read(bytes, 0, count)) > 0)
+            while ((numBytes = input!.Read(bytes, 0, count)) > 0)
```

Comment:
I only see one caller to Pump.  Since we'll throw if `input` is null, it shouldn't be declared as null-accepting.  At worst, we're just moving this `!` to the caller where they're asserting it's not null.

Digging deeper, the `input` parameter can never be null (which is good, or we'd have a pending exception here).

Reply:
Good spot. I've done some of the other suggestions above, and will do these when I get a spare minute later in the week. Thanks for your help!

Reply:
I looked at the caller `EncrypedXml::GetCipherValue` and can see that all variations of `cipherData.CipherReference.Uri` are accounted for (is it null? - is it zero length? - is the first character a #? - otherwise throw) - which, as you say, means that it'll not be null. Have changed the input to reflect that.